src/core.h

```diff
@@ -113,6 +113,53 @@ static inline std::mt19937& prng()
     return PRNG;
 }
 
+#if NTTEC_USE_SIMD == SSE4
+#define ALIGN_SIZE 16
+#endif
+
+#if NTTEC_USE_SIMD == AVX2
+#define ALIGN_SIZE 32
+#endif
+
+template <typename T>
+inline T* aligned_allocate(size_t size)
```

Comment:
Maybe we could use [std::align](http://en.cppreference.com/w/cpp/memory/align) to help here.

There is also [aligned_alloc](http://en.cppreference.com/w/c/memory/aligned_alloc), but it's only available in C++17 :slightly_frowning_face: